### PR TITLE
typed errors overhaul in asserter

### PR DESCRIPTION
**Pull request by cindyxkuang**
Fixes # .

### Motivation

adding typed errors improves the debugging experience for developers who can better determine where a particular error is coming from

### Solution

created new asserter/errs package and added variables for each error message in asserter package

### CODE REVIEW

File: asserter/account.go [hunk 1]
```diff
@@ -17,6 +17,8 @@ package asserter
 import (
 	"fmt"
 
+	"github.com/coinbase/rosetta-sdk-go/asserter/errs"
```
Comment (patrick-ogrady) [inline asserter/account.go:20]: nit: call this `github.com/coinbase/rosetta-sdk-go/asserter/errors`

File: asserter/account_test.go [hunk 2]
```diff
@@ -132,7 +134,7 @@ func TestContainsCurrency(t *testing.T) {
 	}
 }
 
-func TestAccoutBalance(t *testing.T) {
+func TestAccountBalance(t *testing.T) {
```
Comment (patrick-ogrady) [inline asserter/account_test.go:137]: ❤️

File: asserter/construction_test.go [hunk 3]
```diff
@@ -148,22 +149,22 @@ func TestConstructionDeriveResponse(t *testing.T) {
...
-			err: errors.New("address cannot be empty"),
+			err: errs.ErrConstructionDeriveResponseAddrEmpty,
 		},
 	}
 
 	for name, test := range tests {
 		t.Run(name, func(t *testing.T) {
 			err := ConstructionDeriveResponse(test.response)
-			assert.Equal(t, test.err, err)
+			assert.True(t, errors.Is(err, test.err))
```
Comment (patrick-ogrady) [inline asserter/construction_test.go:167]: 🚀

File: fetcher/fetcher.go [hunk 4]
```diff
@@ -83,7 +83,7 @@ type Fetcher struct {
 // Error wraps the two possible types of error responses returned
 // by the Rosetta Client
 type Error struct {
-	Err       error        `json:"err"`
+	Err       error        `json:"errs"`
```
Comment (patrick-ogrady) [inline fetcher/fetcher.go:86]: nit: I'd leave this as `err`

**Comment by patrick-ogrady** [comment]
@cindyxkuang Can we add a method to `errs` called `AsserterErr` that returns true if the provided `error` is an `asserter` error? (should just iterate over all errors on `errors.Is`?

File: asserter/errors.go [hunk 5]
```diff
@@ -0,0 +1,392 @@
...
+		ErrConstructionCombineRequestIsNil,
+		ErrConstructionCombineRequestUnsignedTxEmpty,
+		ErrConstructionHashRequestIsNil,
+		ErrConstructionHashRequestSignedTxEmpty,
+		ErrConstructionParseRequestIsNil,
+		ErrConstructionParseRequestEmpty,
+	}
+)
+
+// IsAssertError takes an error as an argument and returns
```
Comment (patrick-ogrady) [inline asserter/errors.go:363]: nit: should be `// ErrAsserter`

File: asserter/errors.go [hunk 6]
```diff
@@ -0,0 +1,392 @@
...
+// whether or not the error is one thrown by the asserter
+// along with the specific source of the error
+func ErrorAsserter(err error) (bool, string) {
```
Comment (patrick-ogrady) [inline asserter/errors.go:366]: nit: `ErrAsserter`

File: asserter/errors_test.go [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+package asserter
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestIsAssertError(t *testing.T) {
```
Comment (patrick-ogrady) [inline asserter/errors_test.go:24]: nit: `TestErrAsserter`

Comment (cindyxkuang) [reply asserter/errors_test.go:24]: whoops I renamed the function but forgot to rename the test

File: fetcher/fetcher.go [hunk 8]
```diff
@@ -83,7 +83,7 @@ type Fetcher struct {
 // Error wraps the two possible types of error responses returned
 // by the Rosetta Client
 type Error struct {
-	Err       error        `json:"err"`
+	Err       error        `json:"error"`
```
Comment (patrick-ogrady) [inline fetcher/fetcher.go:86]: nit: `err`

**Pull request merged**